vesc_driver/src/vesc_interface.cpp

```diff
@@ -127,15 +133,31 @@ void VescInterface::Impl::rxThread()
     }
 
     // attempt to read at least bytes_needed bytes from the serial port
-    int bytes_to_read = std::max(bytes_needed, 4096);
+    int bytes_to_read = std::min(bytes_needed, 4096);
 
     {
-      std::lock_guard<std::mutex> lock(serial_mutex_);
+      // removed temporarily because it causes a deadlock
+      //std::lock_guard<std::mutex> lock(serial_mutex_);
+      boost::system::error_code ec;
 
       const size_t bytes_read = boost::asio::read(
         serial_port_,
-        boost::asio::buffer(buffer, buffer.size()),
-        boost::asio::transfer_exactly(bytes_to_read));
+        boost::asio::buffer(bufferRx, bufferRx.size()),
+        boost::asio::transfer_exactly(bytes_to_read),
+        ec
+        );
```

Comment:
Instead of doing this directly with `boost::asio`, the idea was to replace the calls with `transport_drivers::serial_driver` and let it do the underlying work. Are you one of the students working with Jack at UCSD or is this just an independent fix?

Reply:
this is an independent fix. i am not a student but a hobbyist who created a robot. the modification I proposed is part of the code I created for a robot composed of 4 vescs connected via canbus. I don't use the node directly (except for single tests) but through a further implementation that incorporates it in ros_control. now i am trying to migrate it to ros2. I hadn't paid much attention to testing. I check them immediately.

Reply:
Not a problem! If this fix works, I'm happy to accept it. I never finished the `boost::asio` implementation but I'm glad to see it had some value. If you handle the other issues, I will accept the PR.